src/allmydata/grid_manager.py

```diff
@@ -0,0 +1,348 @@
+import sys
+import json
+from datetime import (
+    datetime,
+    timedelta,
+)
+
+from allmydata.crypto import (
+    ed25519,
+)
+from allmydata.util import (
+    fileutil,
+    base32,
+)
+
+
+class _GridManagerStorageServer(object):
+    """
+    A Grid Manager's notion of a storage server
+    """
+
+    def __init__(self, name, public_key, certificates):
+        self.name = name
+        self._public_key = public_key
+        self._certificates = [] if certificates is None else certificates
+
+    def add_certificate(self, certificate):
+        self._certificates.append(certificate)
+
+    def public_key(self):
+        return ed25519.string_from_verifying_key(self._public_key)
+
+    def marshal(self):
+        return {
+            u"public_key": self.public_key(),
+        }
+
+
+def create_grid_manager():
+    """
+    Create a new Grid Manager with a fresh keypair
+    """
+    private_key, public_key = ed25519.create_signing_keypair()
+    return _GridManager(
+        ed25519.string_from_signing_key(private_key),
+        {},
+    )
+
+
+def load_grid_manager(config_path, config_location):
+    """
+    Load a Grid Manager from existing configuration.
+
+    :param FilePath config_path: the configuration location (or None for
+        stdin)
+
+    :param str config_location: a string describing the config's location
```

Comment:
From this docstring, I don't understand why this is a different thing from `config_path`.  From the implementation I see it's probably to support better error reporting for the `--config -` case?  Maybe the caller is in a better position to be able to describe this and `load_grid_manager` should only raise an exception and let the caller deal with formatting it as text to a user.  This would make sense factoring-wise, too, since `load_grid_manager` is part of the "model".  The click-based code (or a future GUI or HTTP API or even a programmatic Python user) will have a better idea about how to report it to a person.